akka-stream/src/main/scala/akka/stream/impl/streamref/SourceRefImpl.scala

```diff
@@ -183,17 +183,20 @@ private[stream] final class SourceRefStageImpl[Out](
       }
 
       def tryPush(): Unit =
-        if (receiveBuffer.nonEmpty) push(out, receiveBuffer.dequeue())
+        if (receiveBuffer.nonEmpty && isAvailable(out)) push(out, receiveBuffer.dequeue())
         else if ( /* buffer is empty && */ completed) completeStage()
```

Comment:
now it could also be `receiveBuffer.nonEmpty && completed` when we get here, I think that's wrong

Reply:
Thanks, good catch -- fixed.